Disable Experimental features in Lockdown Mode

brentfulgham opened the pull request:
#### 01a79e657788f8f5f9c29648b0c07d428ec55379
<pre>
Disable Experimental features in Lockdown Mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=248915">https://bugs.webkit.org/show_bug.cgi?id=248915</a>
&lt;rdar://101969455&gt;

Reviewed by Chris Dumez.

Ensure that experimental features (which are generally off by default) are truly off in Lockdown Mode.
This is meant to protect users that might have an experimental feature on for development work,
who would not want that feature active for Lockdown Mode.

* Source/WebCore/Scripts/GenerateSettings.rb: Add new generators to set turn off all unstable features.
* Source/WebCore/Scripts/SettingsTemplates/Settings.cpp.erb: Ditto.
* Source/WebCore/Scripts/SettingsTemplates/Settings.h.erb: Ditto.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::adjustSettingsForLockdownMode): Ensure that all unstable features are off.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm: Add tests.

Canonical link: <a href="https://commits.webkit.org/257758@main">https://commits.webkit.org/257758@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0856fefa3d7836cd6322b900225b7e2aa25ee499

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99893 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9065 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32975 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109246 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169485 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/103895 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/9896 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/86358 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92345 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107153 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/105659 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7477 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34237 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89385 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22181 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2870 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23693 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2822 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/46068 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5325 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/8955 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43166 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4678 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm:
```diff
@@ -7848,6 +7848,20 @@ static void checkSettingsControlledByLockdownMode(WKWebView *webView, ShouldBeEn
+
+    // Confirm unstable Experimental settings are always off in Lockdown Mode.
+    // They are expected to always be off:
+    EXPECT_EQ(runJSCheck("CSS.supports('contain-intrinsic-size: 10rem')"_s), false);
```
I imagine these checks would pass even without your change?

Shouldn't we enable all experimental features at the beginning of the test to actually make sure your code change is tested?

Something like:
```
    for (_WKExperimentalFeature *feature in [WKPreferences _experimentalFeatures])
            [[configuration preferences] _setEnabled:YES forExperimentalFeature:feature];
```

brentfulgham replied:
We actually already turn some of them on in `configureLockdownWKWebViewConfiguration` in the same file. I just need to change that setup to turn them ALL on.

cdumez reviewed (approved):
LGTM if the bots are happy and ideally with the change I suggested (assuming my assumption is correct).

Commit:
Disable Experimental features in Lockdown Mode
https://bugs.webkit.org/show_bug.cgi?id=248915
<rdar://101969455>

Reviewed by Chris Dumez.

Ensure that experimental features (which are generally off by default) are truly off in Lockdown Mode.
This is meant to protect users that might have an experimental feature on for development work,
who would not want that feature active for Lockdown Mode.

* Source/WebCore/Scripts/GenerateSettings.rb: Add new generators to set turn off all unstable features.
* Source/WebCore/Scripts/SettingsTemplates/Settings.cpp.erb: Ditto.
* Source/WebCore/Scripts/SettingsTemplates/Settings.h.erb: Ditto.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::adjustSettingsForLockdownMode): Ensure that all unstable features are off.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ProcessSwapOnNavigation.mm: Add tests.

Canonical link: https://commits.webkit.org/257758@main
